src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,166 @@
+package duke;
+
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.task.Deadline;
+import duke.task.Event;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import java.time.LocalDate;
+
+/**
+ * Handler to load tasks from the file and save tasks into the file
+ */
+public class Storage {
+
+    private String filePath;
+
+    /**
+     * Constructs a Storage object
+     * @param filePath Location of the file to save the tasks into
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        ensureFolderExists();
+        ensureFileExists();
+    }
+
+    /**
+     * Ensures that the given folder of the filePath exists, and creates the folder if it doesn't
+     */
+    private void ensureFolderExists() {
+        File folder = new File(filePath).getParentFile();
+        if (!folder.exists()) {
+            folder.mkdirs();
+        }
+    }
+
+    /**
+     * Ensures that the given file of the filePath exists, and creates file if it doesn't
+     */
+    private void ensureFileExists() {
+        File file = new File(filePath);
+        if (!file.exists()) {
+            try {
+                file.createNewFile();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
```

Comment:
Maybe consider refactoring this part of the code using the guard clause to make the happy path (in this case, the file.createNewFile() statement, more prominent? This can make the code less nested!
```suggestion
    private void ensureFileExists() {
        File file = new File(filePath);
        if (file.exists()) {
            return;
        }
        try {
             file.createNewFile();
        } catch (IOException e) {
             e.printStackTrace();
        }
    }
```